eclipse-collections/src/main/java/org/eclipse/collections/impl/set/sorted/immutable/AbstractImmutableSortedSet.java

```diff
@@ -524,4 +524,35 @@ public int detectLastIndex(Predicate<? super T> predicate)
     {
         throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".detectLastIndex() not implemented yet");
     }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj == this)
+        {
+            return true;
+        }
+
+        if (!(obj instanceof Set))
+        {
+            return false;
+        }
+        Set<?> otherSet = (Set<?>) obj;
+        if (otherSet.size() != this.size())
+        {
+            return false;
+        }
+        try
+        {
+            return this.containsAll(otherSet);
+        }
+        catch (ClassCastException ignored)
+        {
+            return false;
+        }
+        catch (NullPointerException ignored)
+        {
+            return false;
+        }
+    }
```

Comment:
Why does this catch exceptions? Can we have hashcode here too?

Reply:
@motlin some of the implementations of equals that rely on `Collection.containsAll` do catch those exception because `Collection.containsAll` document them and if they are thrown, it's not an error, it means the element are not contained (because of different types or things like that).

I can add `hashCode`, I didn't because I expected that all subclass should be more relevant to implement it, but if I add it, I'm not sure what is the best default implementation? An `hashCode` of all elements?

Reply:
@motlin concerning hashCode, actually findbugs is complaining about it, so I moved it to the two subclasses that needed it.